geometry/optimization/graph_of_convex_sets.cc

```diff
@@ -908,6 +930,7 @@ MathematicalProgramResult GraphOfConvexSets::SolveShortestPath(
   // Implements the rounding scheme put forth in Section 4.2 of
   // "Motion Planning around Obstacles with Convex Optimization":
   // https://arxiv.org/abs/2205.04422
+  bool has_paths_from_source_to_target = true;
```

Comment:
I'm a bit confused by this: are we asking if the discrete graph has a path? Because if the convex relaxation of the SPP in GCS is feasible then, I believe, that the graph must have a path (since there is a unit of flow that somehow goes from s to t). In other words, does it actually happen in practice that the depth-first search backtracks to the source? I would imagine that that cannot happen if the convex relaxation is feasible, while if the convex relaxation is infeasible we should not do any rounding?

Or is this because we want a rounding function that does not make any assumptions on where the flows come from?